go/config/config.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 The Board of Trustees of the Leland Stanford Junior University
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package config
+
+import (
+	"os"
+	"path"
+
+	"gopkg.in/yaml.v3"
+)
+
+// AlmondConfig with almond-cloud configs and secrets
+type AlmondConfig struct {
+	Config Config
+	Secret Secret
+}
+
+// Config file for almond-cloud
+type Config struct {
+	NLServerURL string `yaml:"NL_SERVER_URL"`
+}
+
+// Secret file for almond-cloud
+type Secret struct {
+	DatabaseURL string `yaml:"DATABASE_URL"`
+}
+
+var almondConfig AlmondConfig
+
+func GetAlmondConfig() *AlmondConfig {
+	return &almondConfig
+}
+
+func InitAlmondConfig() error {
+	configDir := os.Getenv("ALMOND_CONFIG_DIR")
+	if len(configDir) == 0 {
+		configDir = "/etc/almond-cloud/config.d"
+	}
+	return ParseAlmondConfig(configDir, &almondConfig)
+}
+
+// ParseAlmondConfig from a directory. Assumes directory contains onfig.yaml and secret.yaml
+func ParseAlmondConfig(dirPath string, almondConfig *AlmondConfig) error {
```

Comment:
Technically, there can be multiple files inside /etc/almond-cloud/config.d, we should iterate all of them, and look for files with extension .yaml or .json.

Also, there is no guarantee that something is defined in config.yaml or secret.yaml, it should all go in the same bag, and it should have defaults defined (for NL_SERVER_URL) in case the value is missing entirely.